packages/utils/src/conversions.ts

```
@@ -0,0 +1,202 @@
+/**
+ * Converts a hexadecimal string to a bigint, ensuring the hex string starts with '0x'.
+ * @param hex The hexadecimal string to convert.
+ * @returns The bigint representation of the hexadecimal string.
+ */
+export function beHexadecimalToBigint(hex: string): bigint {
+    // Ensure the hex string starts with '0x'.
+    const formattedHexString = hex.startsWith("0x") ? hex : `0x${hex}`
+
+    return BigInt(formattedHexString)
+}
+
+/**
+ * Converts a hexadecimal string to a bigint. Alias for beHexadecimalToBigint.
+ * @param hex The hexadecimal string to convert.
+ * @returns The bigint representation of the hexadecimal string.
+ */
+export function hexadecimalToBigint(hex: string): bigint {
+    return beHexadecimalToBigint(hex)
+}
+
+/**
+ * Converts a buffer to a bigint by interpreting the buffer as a hexadecimal string.
+ * @param b The buffer to convert.
+ * @returns The bigint representation of the buffer's contents.
+ */
+export function beBufferToBigint(b: Buffer): bigint {
+    return BigInt(`0x${b.toString("hex")}`)
+}
+
+/**
+ * Converts a buffer to a bigint. Alias for beBufferToBigint.
+ * @param b - The buffer to convert.
+ * @returns The bigint representation of the buffer's contents.
+ */
+export function bufferToBigint(b: Buffer): bigint {
+    return beBufferToBigint(b)
+}
+
+/**
+ * Converts a bigint to a buffer, assuming big-endian byte order, and fills with zeros if necessary.
+ * @param n The bigint to convert.
+ * @param length The number of bytes of the buffer to return.
+ * @returns The buffer representation of the bigint.
+ */
+export function beBigintToBuffer(n: bigint, length?: number): Buffer {
+    const hex = bigIntToHexadecimal(n)
+
+    // Calculate the minimum buffer length required to represent 'n' in bytes.
+    // Each hexadecimal character represents 4 bits, so 2 characters are 1 byte.
+    const minLength = Math.ceil(hex.length / 2)
+
+    // Use the provided length or the calculated minimum length, whichever is greater.
```

This behavior needs to be documented or it'll be surprising.  Perhaps the argument should be called minLenth?
Alternatively if minLength > length this function could throw an exception.  Incorrect buffer lengths can be a source of security bugs, so I think checking and failing loudly might be the best approach.  (The current approach is already much safer than truncating, though.)